### PR TITLE
S2 at

### PR DESCRIPTION
**Pull request by AlisonYoon**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -110,4 +110,23 @@ function currentFilter(string $category):string {
         <li><a class=\"produce\" href=\"index.php?category=produce\">Produce</a></li>
         <li><a class=\"other\" href=\"index.php?category=other\">Other</a></li>");
     }
-}
\ No newline at end of file
+}
+
+/**
+ * inputValidation function takes string$item, string$category, int$price, int$remaining (all should come from $_GET) and return either a INSERT query or error message.
```
Comment (FelixLetheren) [inline functions.php:116]: specifies what the function takes but not what it does with it.

**Review by FelixLetheren — CHANGES_REQUESTED** [review]
Just one comment on docblock

**Review by FelixLetheren — APPROVED** [review]

File: functions.php (hunk 1 above)
Comment (sjr-mayden) [reply functions.php:116]: Agreed

File: index.php [hunk 2]
```diff
@@ -2,13 +2,17 @@
 
 require_once 'functions.php';
 
+$item = (isset($_GET['item'])) ? $_GET['item'] : NULL;
```
Comment (sjr-mayden) [inline index.php:5]: If these are null what happens?

File: index.php [hunk 3]
```diff
@@ -66,14 +70,22 @@
             <input type="text" name="price" placeholder="price (pence)" required>
             <input type="text" name="remaining" placeholder="remaining (%)" required>
             <input type="submit" value="Submit">
+            <p><?php
+                if($item && $category && $price && $remaining) {
```
Comment (sjr-mayden) [inline index.php]: Looks like this fails silently instead of providing an error message, is that coming later?

**Review by sjr-mayden — CHANGES_REQUESTED** [review]

File: index.php (hunk 2 above)
Comment (AlisonYoon) [reply index.php:5]: nothing happens

File: index.php (hunk 3 above)
Comment (AlisonYoon) [reply index.php]: it does throw type error, when wrong data type is passed. When input is empty and submitted, input itself will ask you to fill it in

File: index.php (hunk 2 above)
Comment (sjr-mayden) [reply index.php:5]: Ok cool, so there should be some sort of error/feedback to the user in addition to not inserting into the database

**Review by sjr-mayden — CHANGES_REQUESTED** [review]

File: index.php [hunk 4]
```diff
@@ -46,6 +48,19 @@
                         <input type="text" name="price" placeholder="price (pence)" required>
                         <input type="text" name="remaining" placeholder="remaining (%)" required>
                         <input type="submit" value="Submit">
+                        <p>
+                            <?php
+                            if(!is_null($item) && $category && !is_null($price)  && !is_null($remaining)) {
+                                if((int)$price > 0) {
```
Comment (sjr-mayden) [inline index.php]: Shouldn't the % also be validated the same way?

**Review by sjr-mayden — CHANGES_REQUESTED** [review]

**Review by sjr-mayden — APPROVED** [review]

**Pull request merged**